app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt

```diff
@@ -96,10 +96,8 @@ class NewsFragment @Inject constructor() : WolmoFragment<NewsPresenter>(), INews
         newsAdapter.setUserId(userId)
     }
 
-    override fun onItemNewClicked(new: New) {
-        val intent = Intent(requireActivity(), NewDetailActivity::class.java)
-        intent.putExtra(NEW, new)
-        startActivity(intent)
+    override fun onItemNewClicked(bundle: Bundle, extras: FragmentNavigator.Extras) {
+        Navigation.findNavController(view!!).navigate(NAVIGATION_TO_NEW_DETAIL, bundle, null, extras)
```

Comment:
We are going to do use some Kotlin magic here. We are going to create a separate file called NavigationExtensions.kt. There we will define the following:

`fun Fragment.requireView() = view!!

fun Fragment.requireNavController() = Navigation.findNavController(requireView())`

So, you can avoid doing` Navigation.findNavController(view!!)` and replace it with `requireNavController()`